src/helpers/migration/metadata.js

```diff
@@ -0,0 +1,170 @@
+import log from 'cozy-logger'
+
+import { APP_SETTINGS_DOCTYPE, FILES_DOCTYPE } from 'src/constants'
+import {
+  buildAppSettingQuery,
+  buildFilesFromDateQuery
+} from 'src/helpers/migration/queries'
+
+const OLD_METADATA_ATTRIBUTES = [
+  'cafFileNumber',
+  'cardNumber',
+  'vinNumber',
+  'ibanNumber',
+  'passportNumber'
+]
+const NEW_METADATA_ATTRIBUTE = 'number'
+
+/**
+ * Fetch the io.cozy.mespapiers.settings document
+ *
+ * @param {object} client - The CozyClient instance
+ */
+export const fetchAppSetting = async client => {
+  log('info', 'Start fetchAppSetting')
+  const appSettingQuery = buildAppSettingQuery()
+
+  return client.query(appSettingQuery.definition)
+}
+
+/**
+ * Fetch the files indexed on their updatedAt date and
+ * the existence of the "metadata" property
+ *
+ * @param {object} client - The CozyClient instance
+ * @param {string} date - The starting date to query
+ * @param {number} limit - The maximum number of files by request
+ */
+export const fetchFilesFromDateWithMetadata = async ({
+  client,
+  date,
+  limit
+}) => {
+  log('info', 'Start fetchFilesFromDate')
+  const filesFromDateQuery = buildFilesFromDateQuery(date, limit)
+
+  return client.queryAll(filesFromDateQuery.definition)
+}
+
+/**
+ * Update IOCozyMespapiersSettings with the last process date
+ *
+ * @param {object} client - The CozyClient instance
+ * @param {object} appSettings - The io.cozy.mespapiers.settings document
+ * @param {string} lastProcessedFileDate - The last processed file date
+ */
+export const updateAppSettings = async ({
+  client,
+  appSettings,
+  lastProcessedFileDate,
+  lastRunningMigrateMetadataService
+}) => {
+  if (appSettings) {
+    return client.save({
+      ...appSettings,
+      lastProcessedFileDate:
+        lastProcessedFileDate || appSettings.lastProcessedFileDate,
+      lastRunningMigrateMetadataService:
+        lastRunningMigrateMetadataService ||
+        appSettings.lastRunningMigrateMetadataService
+    })
+  } else {
+    return client.create(APP_SETTINGS_DOCTYPE, {
```

Comment:
je crois qu'on n'utilise plus create/update en direct, mais uniquement save qui s'occupe de faire l'un ou l'autre, l'avantage étant qu'on est sûr avoir le même type de retour

Reply:
> l'avantage étant qu'on est sûr avoir le même type de retour

Ça tu l'as aussi avec create / save / update sans souci, c'est le même chemin de code et heureusement que ça retourne la même chose. 

> je crois qu'on n'utilise plus create/update en direct

Si vous n'avez pas besoin des `references` alors create reste la bonne solution. https://github.com/cozy/cozy-client/blob/master/packages/cozy-client/src/CozyClient.js#L628 

Ici ça semble ne pas être le cas, donc oui ici on peut virer ce if et ne faire que le save() 

Pour info, la doc getting started de cozy-client est bien à jour https://github.com/cozy/cozy-client/blob/master/docs/getting-started.md#mutate-the-data

Reply:
Merci pour le rappel, j'ai changé le code pour ne passer que par un save.